Fix home

hgouraud opened the pull request:
use Perso.interp_templ rather than Util.open_etc_file
The latter does not interpret (or remember) the %let commands found in the template file.
This modification removes multiple errors appearing when the home.txt template file was called (undefined rng, random).
The fix has not been applied to Notes display as the base parameter is not available in this context.
For another case, the use of Perso.interp_templ created a dependency cycle!

hgouraud committed:
display home as a full template (corrects rng and random errors)

a2line commented:
Première remarque : j'ai des erreurs à la pelle en visiteur quand je ne suis pas connecté, alors que ça a l'air fonctionnel en magicien.

a2line committed:
fix modify place translations

hgouraud committed:
remove home from perso_header; correct SND_IMAGE_C_OK in menubar

a2line committed:
safer snd/del_image buttons

hgouraud committed:
return 0 rather that '' for %family_cnt

a2line committed:
use double bar for Sosa

add managment of double height when a grandparents' couple is divorced or separated
factorize individual data

hgouraud committed:
add link to sosa_ref in search pop-up

hgouraud committed:
suppress Hutil_2

michelmno commented:
I tested last update of this PR with cid b74350e66, and for some individual with no parents, the error message is switching from `unbound var "sosa_ref.key"` already tracked by issue #1778 to warning `must test "has_parents"` as extracted below.
```
[2024-04-23 05:50:43]: WARNING ./etc/modules/arbre_3gen_photo.txt 1447-1447: since v5.00, must test "has_parents" before using "mother"
[2024-04-23 05:50:48]: WARNING ./etc/modules/arbre_3gen_photo.txt 1385-1385: since v5.00, must test "has_parents" before using "father"
```

hgouraud commented:
thanks for the test. fixed in c03357e61

hgouraud commented:
I tried to build some sort of regression test.
Can be expanded by adding additional commands
Look at test/run-GW-test.sh
You have to "personalize" according to your own base for some of the commands
Tell me what you think, and of possible improvements

michelmno commented:
I am suggesting a script change in https://github.com/michelmno/geneweb/commit/146f10af49a7c09866be684f188c152053a0d045

michelmno commented:
first run with my DB is reporting: 
```
[2024-04-24 07:34:28]: WARNING ./etc/buttons_dag.txt 6547-6548: since v5.00, must test "has_parents" before using "mother"
[2024-04-24 07:34:28]: WARNING ./etc/buttons_dag.txt 6564-6565: since v5.00, must test "has_parents" before using "mother"
[2024-04-24 07:34:28]: WARNING ./etc/buttons_dag.txt 8178-8179: since v5.00, must test "has_parents" before using "father"
[2024-04-24 07:34:28]: WARNING ./etc/buttons_dag.txt 8195-8196: since v5.00, must test "has_parents" before using "father"
```

a2line commented:
Pleas DON’T merge this PR. It is draft. I see a lot of mistakes to be fixed: double titles + double HTML headers on some .ml pages, menubar removed from carousel or not working at all on some old pages…

a2line committed:
lot of minor template fixes

michelmno commented:
another error detected by `run-GW-test.sh`  when invalid p value
`http://localhost:2317/testbasefusionnee?m=FORUM&p=939`
generating a critical error
`[2024-04-26 08:46:51]: CRITICAL testbasefusionnee_w?m=FORUM&p=939 Invalid_argument("rseek_in")`
Should I create a specific issue ?

hgouraud commented:
yes. In fact no, this test is too base specific and the failure is inevitable.
I am pushing an update for run-GW-test following your modifications

hgouraud committed:
remove calls to perso_header in ml code

hgouraud committed:
provide name of target for shortest path

fix #1782

hgouraud committed:
navigate with Sosa across ? ?

hgouraud committed:
fix plugin selection.

fix #1788

hgouraud committed:
no call to home in gwd.ml

hgouraud committed:
fix “marié ave”

a2line committed:
add notes and sources in HTML tooltips for first NBMDS events

new bvar "event_tooltips=no" to hide those new tooltips
few more template fixes

michelmno commented:
@hgouraud, I have new changes proposal for test/run-GW-test.sh in cid https://github.com/michelmno/geneweb/commit/e514bdd44513c0dfb0b3b5484b0dd68f18e54c52

Using it in my environment is reporting CRITICAL errors similar to issue #1788 when url parameters are not valid. But also some unbound failures as detailed in attached log 
[gwd_cgi_tst_error.extract.log](https://github.com/geneweb/geneweb/files/15142622/gwd_cgi_tst_error.extract.log)

hgouraud commented:
I am working ion it too. let me push my current status
I will make it a new branch (test-gw) as A2 is also working on this branch

hgouraud committed:
add missing hd/etc/accent.txt (link into templm)

a2line committed:
add a button to move union comment to marriage notes

a2line committed:
add missing htrees module

michelmno commented:
> I am working ion it too. let me push my current status I will make it a new branch (test-gw) as A2 is also working on this branch

I follow your new branch in  https://github.com/michelmno/geneweb/commits/test-gw/ with updated cid
https://github.com/michelmno/geneweb/commit/18f766d25f7e5bf70efbaefa8f2d257c7ab8d3cb

hgouraud commented:
@michelmno : I manually merged your proposed changes in fix-home
Thanks

hgouraud committed:
behave when plugins is empty in .gwf

hgouraud commented:
@michelmno: can you propose a way to have `test-gw-vars.txt` as default value for `-f`

michelmno commented:
> @michelmno: can you propose a way to have `test-gw-vars.txt` as default value for `-f`

Where is it supposed to be ? in same folder as test/run-GW-test.sh ?

michelmno commented:
> I tried to build some sort of regression test. Can be expanded by adding additional commands Look at test/run-GW-test.sh You have to "personalize" according to your own base for some of the commands Tell me what you think, and of possible improvements

For automation, you do not need a default  -f parameter
I would suggest  a call to test/run-GW-test.sh  -f <FQ_path/test-gw-vars.txt>  <basename>
Where <FQ_path/test-gw-vars-basename.txt> is the choice of caller.
We could add a var in test-gw-vars.txt to specify the basename, so that <basename> not to be mandatory
There is no easy way to define a default test-gw-vars.txt location

I will continue to discuss later on, I have to leave

hgouraud commented:
I would put a default test-gw-vars.txt file in the geneweb GitHub repo where it could stay as an untracked file.

michelmno commented:
@hgouraud, I just pushed two commits https://github.com/michelmno/geneweb/commit/974530d4acbedf2af4c5404e8a35cbd363808c34 & https://github.com/michelmno/geneweb/commit/40c0e31f625ab50b69463602036fc8cfb54e9ece you may cherry pick in your hgouraud/fix-home branch

hgouraud committed:
perso_modules .gwf -> perso_utils

hgouraud committed:
run-gw-test.sh: some sort of regression test

Thanks to Michelmno for testing and suggestions.

hgouraud committed:
explicit messages for incorrect request

hgouraud committed:
handle empty list

hgouraud committed:
suppress trace message

hgouraud committed:
add ?(no_title = false) parameter to Hutil.header_with_title

hgouraud committed:
Hutil.header_with_title -> Hutil.header

hgouraud committed:
marriage_date returns null_val rather then raise Not_found

hgouraud committed:
suppress link_to_welcome

hgouraud committed:
handle sosa_ref in templ.ml

hgouraud committed:
search person as pp, np, ocp as well as ip. Explicit incorrect request message

a2line committed:
add a button to linked pages on menubar with count

*Add a similar image counter on carrousel button of menubar.
*New template variable %linked_pages_nbr; prints the number of linked pages.
*Full revamp of m=LINKED page using a table structure instead a list to improves readability.
*Remove duplicate home/referer and all tt HTML tags.
*Add specific icons and a description column for the five cases.
*Add a function to manage absent, empty and whitespaces only note titles (TITLE=).

*TODO: Manage notes of all v7 p/f_events!

a2line committed:
fix critical error with pt-br in url_set

parser is failing with the hyphen seen as a minus operator

hgouraud committed:
trying to hancle BCE (negative) dates

michelmno committed:
run-GW-test.sh Check bvar in gwf to only test cmd if possible

This is required to avoid useless error
if database is not configured for related command.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
suppress link_to_welcome in modInd and modFam

hgouraud committed:
bad chars encoding

hgouraud reviewed (commented):
Indeed!

michelmno committed:
run-GW-test.sh to exit with rc=1 only if failure

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Add missing plugins bvar in etc/a.gwf

Was an already usable bvar since GW version 7.0
but never added in this reference file.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

a2line committed:
add divorce/separation indication on right border in tables

a2line committed:
fix broken miss_evt table, add only=on support

*Fix lvl 0 logic.
*Fix page header and translations of hyperlink titles.
*Remove useless “nbsp” trick for fake empty cells.
*Remove background style= and table align=, use CSS instead.
*Now only one background color for missing event, same as in full tables.
*Fix depreciated name= to id= in non closed a tag.

*Minor partial cleanup in ancsosa/anclist.

hgouraud committed:
managing divorce and separation as events

rewrote get_divorce to ensure that divorce information comes from Efam_Divorce fevent
added get_separation similar to get_divorce returning Separated d
removed output of divorce/separation (- d/#sep) in gwu as they will be treated as fevents
added various separation commands similar to divorce in perso.ml
adjusted unions.txt template accordingly

hgouraud committed:
in gwu keep old divorce and separation behaviour if old_gw

hgouraud committed:
added NotSeparated value to divorce field

hgouraud committed:
in updateFam.ml, obtain divorce/separation from events rather than from fam.divorce

a2line committed:
better table header/tooltips

factorize header + place/date_popup functions in new table_utils subtemplate
fix headers for Firefox
simpler conditions for tooltip (wip)

a2line committed:
change input type to password

fix #1796

hgouraud commented:
@canonici, @ilankri, can you review? This is not WIP anymore

hgouraud committed:
fix duplicate trl; fix back-arrow in notesDisplay

a2line committed:
add a new ± button to prevent too quick high generation computing

* A new javascript function hide level btn > 12 (hard fixed atm,
but quite strange if max=13…), perhaps we need to remove the hiding "-" button,
it's not really usefull and can take a line for itself in some cases!
* Load once js function to copy the caution in title of every high v buttons.
* Hide the whole generation selector toolbar when not needed.
* Fix selected v in toolbar when evar.v > max_ancestor_level
and also e_v value for that case.

Also:
* Fix some rendering for some cases (fixing max_anc).
* Remove gen in title and fix left margin vs container-fluid.
* Fix him/her box, add a translation for neutral sex (iel in fr).
* Remove generation reminder (big numbers on left) if v<4.
* Add minimalist tooltips for each statistics.
* Rollback menubar default v= to 7 (was downgraded to 3).

a2line committed:
prefer bold to white to improve readabilty when console background is white

hgouraud committed:
read #sep without date in family description

a2line committed:
few adjustement

a2line committed:
remove not found error on Windows

a2line committed:
fix gradients, allow sorting by age

* replace sort functions by a lone polyvalent function
* use order:-1 to sort descreasingly
* sort by age using julian day “ages”
* cleaner code

a2line committed:
add color by couple on ancestor picker, few fixes

* Ajust max_level, print max level on + button tooltip
* Add curlev+1 to print right box-drawing caracteres when individual has
descendant(s) at the hidden v+1 generation (not really estheatical but info was missing)
* Better anc/deslist icons for button bar and menubar.
* Add missing fa-rotate-90 in CSS!
* Fix left margin overlaping home.

a2line committed:
add thousand separator function for template

a2line committed:
fix critical error because of missing %t

a2line committed:
replace png by icons, better table

hgouraud committed:
test for existence of lexicon file. Avoid server crash

hgouraud committed:
suppress link_to_welcome

a2line committed:
fix bad d'Aboville numerotation

Notation is now the same as in destable
Add short title and use monospace font for notation

a2line committed:
add navigation tooltips, improve title and relation information display

fixes level 0 for desc1, also adapt title, add extend possibility to v+ button and remove useless total
display relation date in small, remove bold for married/italic for not_married, switch to X for not_married instead
move id/d-none for filters to the form itself to remove superflous line
center title in ancestor picker modal

a2line committed:
use not-ocamlfind only if camlp5 version >= 8.03.00

Add two new targets to patch/unpatch bin/ged2gwb/dune.in and ged2gwb.ml in
order that all various build work again. The perl regexps are compatible
between all OS (sed one are diverging between Unix and BSD). The modifications
should be transparent for every usages. The "distrib" target has been reworked
a bit to allow “make fmt distrib” used on Github Action.

a2line committed:
fix encoding

my bad!

a2line committed:
enhance navbar, search modal and copyr for small viewports

plus minor other responsive tweaks

hgouraud committed:
split username before using it

hgouraud committed:
list only descendants not all individuals

hgouraud committed:
fix --plugins vs b.plugins

michelmno committed:
Remove perso_module_* from etc/a.gwf

This is to complete PR #1784 and
cid 11f0c8739 "perso_modules .gwf -> perso_utils"

Do a not related change: myname replaced by <basename>
to be homogeneous in a.gwf.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

a2line committed:
better format of commit message and distrib log, make gwd

add a new target “gwd” to build only gwd+gwc executables

michelmno committed:
run-GW-test.sh completed with test of etc/modules/*

This is a basic successive load of etc/modules/* files
to check if gwd is reporting error in generated web page.

Only three use cases:
* existing DBNAME.gwf
* DBNAME.gwf symbolic link to another file
* no existing DBNAME.gwf

Add detection of "Access refused" if $GWCGI script robot_xcl parm

Few cosmetic changes.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
p_mod=1; fix arbre_h6

hgouraud committed:
provide list of modules for test

michelmno committed:
test/run-GW-test.sh Add debug opt & tests based on anctree.txt

* Debug option allow to print out url for manual usage.
* Report nb of errors if any, on perso_modules tests.
* Trial adding more tests as per hd/etc/anctree.txt

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Add new test/test-gwu.sh test script

To compare gwc input and gwu output expecting no changes.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

hgouraud committed:
Merge pull request #16 from michelmno/test-gw

test/run-GW-test.sh Add debug opt & tests based on anctree.txt

ilankri commented:
In the future, no such monster pull request please. Always keep [this](https://github.com/geneweb/geneweb/blob/master/CONTRIBUTING.md) in mind when contributing, thanks!
